### DISCUSSION TITLE
Need Help :)

### DISCUSSION THREAD

**Discussion by mgrossmann** [adesutherland/CMS-370-BREXX#67] [Show and tell]
Hi friends,

I'm just trying to get my beloved rexx repl to work on VM/370.
Trouble starts if I'm calling a host command with *special* arguments:

If anyone has a clue, I would appreciate a hint.

/MIkeG

My current code:

```rexx
 /* BREXX */                         
 PARSE VERSION LANG VER                  
                                
 SAY "### "LANG" "VER" ###"              
 SAY "                      "            
 SAY "TYPE EXIT TO EXIT.    "            
 
 LOOP:                                   
 SAY "Ready(brexx);"                     
 RC   = 0                                
 PARSE PULL CMD                          
 
 SIGNAL ON ERROR                         
 SIGNAL ON SYNTAX                        
 
 INTERPRET CMD                           
 SIGNAL LOOP                             
 
 ERROR:                                  
 SAY '  An error occurred'               
 SIGNAL LOOP                             
 
 SYNTAX:                                 
 SAY '  Unknown command ' CMD            
 SIGNAL LOOP                            

```
This results in: (Same with `l` vs. `l * exec a`)
```
brxrpl                                                                
### CMS bREXX 0.9.12 Mar 15 2021 ###                                  
                                                                      
TYPE EXIT TO EXIT.                                                    
Ready(brexx);                                                         
say Hi folks...                                                       
HI FOLKS...                                                           
Ready(brexx);                                                         
address command                                                       
Ready(brexx);                                                         
identify                                                              
MAINT    AT DRNMIG1A VIA RSCS     10:54:51 03/19/21 GMT     FRIDAY    
Ready(brexx);                                                         
identify help                                                         
DMSIDE070E Invalid parameter HELP.                                    
Ready(brexx);            
identify ( lifo                     
  Unknown command  identify ( lifo  
Ready(brexx);                       
Ready(brexx);                                                                   
'identify ( lifo'                                                               
Ready(brexx);                                                                   
  Unknown command  MAINT    AT DRNMIG1A VIA RSCS     10:55:56 03/19/21 GMT     FRIDAY                                                                           
Ready(brexx); 
```

**Comment by BobBolch** [comment]
I tested on VM/SP and I get the same results as your test
on VM.370 BREXX, except for the last test case:

 brxrpl                                                                    
### REXX370 3.40 17 Jan 1984 ###                                           
                                                                           
TYPE EXIT TO EXIT.                                                         
Ready(brexx);                                                              
 say Hi folks ...                                                          
HI FOLKS ...                                                               
Ready(brexx);                                                              
 address command                                                           
Ready(brexx);                                                              
 identify                                                                  
APL360   AT LCML4381 VIA MIRROR   03/19/21 06:55:30 PDT      FRIDAY        
Ready(brexx);                                                              
 identify help                                                             
Invalid parameter HELP                                                     
  An error occurred                                                        
Ready(brexx);                                                              
 identify ( lifo                                                           
  Unknown command  identify ( lifo                                         
Ready(brexx);                                                              
                                                                           
Ready(brexx);                                                              
 'identify ( lifo'                                                         
    16 *-*  'identify ( lifo'                                              
       +++ RC(-3) +++                                                      
  An error occurred                                                        
Ready(brexx);                                                              
 'IDENTIFY ( LIFO'                                                         
Ready(brexx);                                                              
  Unknown command  APL360   AT LCML4381 VIA MIRROR   03/19/21 06:56:34 PDT 
FRIDAY                                                                     
Ready(brexx);

**Reply by mgrossmann** [reply]
Thanks for testing, Bob.

Do you have any idea to improve my exec? Or may this be an limitation of REXX interpret function?

/MIkeG

**Comment by BobBolch** [comment]
The left parenthesis in an unquoted string being an error surprised me, but I think it is correct.
A 'PARSE UPPER PULL CMD' could left quoted commands with parenthsis work, but you lose mixed case.

I have something similar that I have used since the old days:

IMMED EXEC to enter REXX immediate mode command processing:


/* Go into REXX IMMEDiate interactive trace mode.              */      
/* REXX statements can be entered interactively.               */      
/* This makes a convenient calculator app for doing hex math.  */      
address COMMAND                                                        
numeric digits 20                                                      
'SET EXECTRAC ON'                                                      
say 'Enter REXX statements:'                                           
Exit 0

**Comment by BobBolch** [comment]
some of the '*' characters did not make it through a cut and paste.

**Reply by mgrossmann** [reply]
thx bob

**Reply by mgrossmann** [reply]
Bob,

I found the problem and are now searching for a solution. :)

```
'IDENTIFY ( LIFO'                                                                 
DBG> 'IDENTIFY ( LIFO'                                                            
Ready(brexx);                                                                     
DBG> MAINT    AT DRNMIG1A VIA RSCS     15:14:54 03/19/21 GMT     FRIDAY           
  Unknown command  MAINT    AT DRNMIG1A VIA RSCS     15:14:54 03/19/21 GMT     F RIDAY                                                                             
Ready(brexx);   
```                                                                  

The `IDENTIFY ( LIFO` host command got executed, as wished. Problem is my eval loop.
The `PARSE PULL CMD` is immediately pulling the result from the stack and tries to execute it. :)

/MIkeG
